src/ResourceManager/Sql/Commands.Sql/Auditing/Cmdlet/AuditingSettings/SetAzureSqlDatabaseAuditingSettings.cs

```diff
@@ -0,0 +1,124 @@
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Sql.Auditing.Model;
+using Microsoft.Azure.Commands.Sql.Common;
+using Microsoft.Azure.Commands.Sql.Services;
+using System;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.Auditing.Cmdlet
+{
+    /// <summary>
+    /// Sets the auditing settings properties for a specific database.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Set, "AzureRmSqlDatabaseAuditing", SupportsShouldProcess = true), OutputType(typeof(DatabaseBlobAuditingSettingsModel))]
+    public class SetAzureSqlDatabaseAuditingSettings : SqlDatabaseAuditingSettingsCmdletBase
+    {
+        /// <summary>
+        /// Gets or sets the state of the auditing policy
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The state of the auditing policy")]
+        [ValidateSet(SecurityConstants.Enabled, SecurityConstants.Disabled, IgnoreCase = false)]
+        [ValidateNotNullOrEmpty]
+        public string State { get; set; }
+
+        /// <summary>
+        ///  Defines whether the cmdlets will output the model object at the end of its execution
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
+
+        /// <summary>
+        ///  Defines the set of audit action groups that would be used by the auditing settings
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The set of the audit action groups")]
+        public AuditActionGroups[] AuditActionGroup { get; set; }
+
+        /// <summary>
+        ///  Defines the set of audit actions that would be used by the auditing settings
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The set of the audit actions")]
+        public string[] AuditAction { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the storage account to use.
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The name of the storage account")]
```

Comment:
One nice thing to do here would be to have a parameter set that accepts an IStorageContext.  This would allow you to pass in values directly from the storage cmdlets.  This is a nice to have, not a requirement